codeguru_profiler_agent/agent_metadata/agent_debug_info.py

```diff
@@ -0,0 +1,70 @@
+from codeguru_profiler_agent.utils.synchronization import synchronized
+
+
+class ErrorsMetadata:
+    def __init__(self):
+        self.errors_count = 0
+        self.sdk_client_errors = 0
+        self.configure_agent_errors = 0
+        self.configure_agent_rnfe_auto_create_enabled_errors = 0
+        self.create_profiling_group_errors = 0
+        self.post_agent_profile_errors = 0
+        self.post_agent_profile_rnfe_auto_create_enabled_errors = 0
+
+    def reset(self):
+        self.errors_count = 0
+        self.sdk_client_errors = 0
+        self.configure_agent_errors = 0
+        self.configure_agent_rnfe_auto_create_enabled_errors = 0
+        self.create_profiling_group_errors = 0
+        self.post_agent_profile_errors = 0
+        self.post_agent_profile_rnfe_auto_create_enabled_errors = 0
+
+    """
+    This needs to be compliant with errors count schema.
+    https://code.amazon.com/packages/SkySailProfileIonSchema/blobs/811cc0e7e406e37a5b878acf31468be3dcd2963d/--/src/main/resources/schema/DebugInfo.isl#L21
+    """
+
+    def serialize_to_json(self):
+        return {
+            "errorsCount": self.errors_count,
+            "sdkClientErrors": self.sdk_client_errors,
+            "configureAgentErrors": self.configure_agent_errors,
+            "configureAgentRnfeAutoCreateEnabledErrors": self.configure_agent_rnfe_auto_create_enabled_errors,
+            "createProfilingGroupErrors": self.create_profiling_group_errors,
+            "postAgentProfileErrors": self.post_agent_profile_errors,
+            "postAgentProfileRnfeAutoCreateEnabledErrors": self.post_agent_profile_rnfe_auto_create_enabled_errors
+        }
+
+    @synchronized
+    def increment_sdk_error(self, error_type):
+        self.errors_count += 1
+        self.sdk_client_errors += 1
+
+        if error_type == "configureAgentErrors":
+            self.configure_agent_errors += 1
+        elif error_type == "configureAgentRnfeAutoCreateEnabledErrors":
+            self.configure_agent_rnfe_auto_create_enabled_errors += 1
+        elif error_type == "createProfilingGroupErrors":
+            self.create_profiling_group_errors += 1
+        elif error_type == "postAgentProfileErrors":
+            self.post_agent_profile_errors += 1
+        elif error_type == "postAgentProfileRnfeAutoCreateEnabledErrors":
+            self.post_agent_profile_rnfe_auto_create_enabled_errors += 1
```

Comment:
why not use a dictionary directly? Those look like the keys that we eventually use to serialize so just store them in a generic way and serialize them as is. You are not currently dealing with unknown arbitrary keys with a proper error so you may as well just send them through as is and it will make it simpler to add a new counter.

also the logic is a bit disturbing to me, I understand you are doing the same as for the java agent but it looks like in the case of a RNF exception in a configure agent, the "configureAgentRnfeAutoCreateEnabledErrors" counter gets +1 as well as "sdkClientErrors" and "errorsCount" but not "configureAgentErrors" which is confusing, it means "configureAgentErrors" really is "otherConfigureAgentErrors" or "unknownConfigureAgentErrors".

Reply:
I agree to the concern of error count mismatch, "configureAgentErrors" should also be incremented in case of RNFE. I updated the code with that change. Thanks for pointing this out.

We can use the dictionary, the only problem is that unless you touch a key in the dictionary it is not going to populate that. So, if lets say we get only `configureAgent` error, then the errors which will not be incremented and added to the dictionary will not be present in the json. This will result in non-continuous emitting of some metrics in the backend. Instead keep it as it is, so that we get to know in the current profile exact number of errors for a particular failure be it 0.